### PR TITLE
SCUMM: replace strcpy calls with Common::strlcpy

### PR DESCRIPTION
**Pull request by athrxx**
This replaces every single strcpy call in the engine with
Common::strlcpy.

Some of these cases might seem a bit pointless, but it is
supposed to discourage future use of strcpy. If people don't
see a single occasion of it they might think twice before they
use it...

I have turned it into a PR so whoever is interested can
check whether he/she sees anything suspicious.

### CODE REVIEW

File: engines/scumm/he/logic/football.cpp [hunk 1]
```diff
@@ -472,7 +472,8 @@ int LogicHEfootball2002::getPlaybookFiles(int32 *args) {
 
 	// Now store the result in an array
 	int array = _vm->setupStringArray(output.size());
-	strcpy((char *)_vm->getStringAddress(array), output.c_str());
+	Common::strlcpy((char *)_vm->getStringAddress(array), output.c_str(), output.size() + 1);
+	_vm->getResourceSize(rtString, array);
```
Comment (bluegr) [inline engines/scumm/he/logic/football.cpp:476]: Is this addition intentional?

Comment (athrxx) [reply engines/scumm/he/logic/football.cpp:476]: Heya, thanks for going over this PR.
I think this is intentional, since ScummEngine_v72he::setupStringArray() also adds +1 to the size argument inside the function. And Common::strlcpy will also reserve one char for the termination `\0`. So, it should work out precisely. But I have to admit this makes my head spin a bit right now. Maybe I just copy the code over to some target that I actually have and test it...

Comment (athrxx) [reply engines/scumm/he/logic/football.cpp:476]: So, I've tested it with Putt-Putt travels through time (just copied over those lines) and it seems fine...

**Comment by bluegr** [comment]
Looks good! The changes are straightforward, and usage of strcpy() is removed (and thus discouraged).
I've played a bit of MI1, MI2, MI3, Sam'n'Max, MM, DOTT, Indy3, Indy4, Zak, Loom, Dig and FT with these changes without any apparent issues.
This has been up for 8 days, without any further comments. Since the changes are straightforward, and there aren't any regressions at a first glance, this an be safely merged.

Merging, thanks again for your work!
